src/server_manager/web_app/ui_components/app-root.js

```diff
@@ -874,6 +933,20 @@ export class AppRoot extends mixinBehaviors
     this.maybeCloseDrawer();
   }
 
+  _sortLanguageNameAscending(a, b) {
+    return a.name > b.name ? 1 : -1;
+  }
+
+  _shouldHideLanguageCheckmark(language, languageCode) {
+    return language !== languageCode;
+  }
+
+  _languageChanged(event) {
+    const code = event.detail.value;
```

Comment:
This should be firing the 'SetLanguageRequested' event instead. You should only need the language code. The handling should know about the direction. The component doesn't display the direction.

Reply:
Done